pydantic/_internal/_typing_extra.py

```diff
@@ -62,6 +63,10 @@ def origin_is_union(tp: type[Any] | None) -> bool:
 if hasattr(typing, 'Literal'):
     LITERAL_TYPES.add(typing.Literal)  # type: ignore
 
+DEPRECATED_TYPES: set[Any] = {deprecated}
+if hasattr(warnings, 'deprecated'):
+    DEPRECATED_TYPES.add(warnings.deprecated)  # type: ignore
```

Comment:
To address https://github.com/pydantic/pydantic/pull/8237/files#r1424563069
```suggestion
# Check if `deprecated` is a type to prevent errors when using typing_extensions < 4.9.0
DEPRECATED_TYPES: tuple[Any, ...] = (deprecated,) if isinstance(deprecated, type) else ()
if hasattr(warnings, 'deprecated'):
    DEPRECATED_TYPES = (*DEPRECATED_TYPES, warnings.deprecated)
```
(this also makes `DEPRECATED_TYPES` a tuple, so you can drop the extra call to `tuple` in the `is_deprecated_instance` function below)